src/js/actions/type.js

```diff
@@ -57,24 +57,55 @@ define(function (require, exports) {
      * @private
      * @param {number} documentID
      * @param {string} name localized name to put into the history state
+     * @param {boolean} modal is the app in a modal state
      * @param {boolean=} coalesce Whether to coalesce this operations history state
      * @return {object} options
      */
-    var _getTypeOptions = function (documentID, name, coalesce) {
-        return {
+    var _getTypeOptions = function (documentID, name, modal, coalesce) {
+        var options = {
             paintOptions: {
                 immediateUpdate: true,
                 quality: "draft"
             },
-            historyStateInfo: {
+            canExecuteWhileModal: true,
+            ignoreTargetWhenModal: true
+        };
+
+        if (!modal) {
+            options.historyStateInfo = {
                 name: name,
                 target: documentLib.referenceBy.id(documentID),
                 coalesce: !!coalesce,
                 suppressHistoryStateNotification: !!coalesce
-            },
-            canExecuteWhileModal: true,
-            ignoreTargetWhenModal: true
-        };
+            };
+        }
+
+        return options;
+    };
+
+    /**
+     * Reset bounds after completing updates to type properties,
+     * but only if we're not in a modal text edit state.
+     *
+     * Note that it is the resetBounds that acts as the history state commit,
+     * a paradigm we'd like to factor eventually.
+     *
+     * @private
+     * @param {Promise} psPromise The promise of the photoshop update operation
+     * @param {Promise} jsPromise The promise of the flux dispatch
+     * @param {Document} document
+     * @param {Immutable.List<Layer>} layers
+     * @param {boolean} modal in modal text edit state?
+     * @return {Promise} Resolves when both input promises are resolved,
+     *                            and after the resetBounds promise, if necessary
+     */
+    var _resetBoundsAfter = function (psPromise, jsPromise, document, layers, modal) {
+        return Promise.join(psPromise, jsPromise,
```

Comment:
I don't like passing the promises into this function. I realize that there is otherwise the duplication of clients needing to `join` the promises on their own. I can't clearly articulate why I don't like this. It just "feels" like the wrong separation for a very small payoff. :walking: